tests/test_base.py

```diff
@@ -607,13 +610,17 @@ def pdf(x, a=1 / np.sqrt(2 * np.pi), x0=0, sigma=1, offset=0):
     with pytest.raises(Exception):
         h.plot_pull({"a": 1})
 
+    plt.close("all")
+
     # wrong kwargs names
     with pytest.raises(Exception):
         h.plot_pull(pdf, abc="crimson", xyz="crimson")
 
     with pytest.raises(Exception):
         h.plot_pull(pdf, ecolor="crimson", mfc="crimson")
 
+    plt.close("all")
```

Comment:
The exceptions still open figures, that's why you were not able to stop this from having 20+ plots. We really should make sure exceptions don't open plots if possible!